bug fix: Bravais Class

galvanixer opened the pull request:
1. Real Space to Coordinates (and vice versa) code modified to properly handle dimensions>2. Earlier, it was not even working for Triangular lattice in 2D.
2. Tests written for GetRealSpace, GetReciprocalSpace(), and GetNeighbor().

galvanixer committed:
Bug Fix: Bravais Class
1. Real Space to Coordinates (and vice versa) code modified to properly handle dimensions>2. Earlier, it was not even working for Triangular lattice in 2D.
2. Tests written for GetRealSpace, GetReciprocalSpace(), and GetNeighbor().

gdmsl commented on test/BravaisTest.cpp:
```diff
@@ -0,0 +1,192 @@
+//===-- HistAccumulatorTest.cpp --------------------------------*- C++ -*-===//
```
```suggestion
//===-- BravaisTest.cpp ----------------------------------------*- C++ -*-===//
```

gdmsl replied:
```suggestion
//===-- BravaisTest.cpp ----------------------------------------*- C++ -*-===//
```

gdmsl commented on test/BravaisTest.cpp:
```diff
@@ -0,0 +1,192 @@
+
+TEST_CASE("Chain Lattice")
+{
+  auto bravais=ChainLattice;
```
Could you run `clang-format` on the files?

gdmsl commented on test/BravaisTest.cpp:
```diff
@@ -0,0 +1,192 @@
+TEST_CASE("Chain Lattice")
+{
+  auto bravais=ChainLattice;
+  vector<Bravais::coords_t> coords{{0},{1},{-2}};
```
Use something like

```
auto coords = vector<Bravais::coords_t>{{0}, {1}, {-2}};
```

Let's have types on the right of assignments.

gdmsl commented on test/BravaisTest.cpp:
```diff
@@ -0,0 +1,192 @@
+  }
+
+  SECTION("GetNeighbor tests"){
+    for (auto idx=0; idx < points.size(); idx++) // points represent neighbors
```
```suggestion
     // points represent neighbors
    for (auto idx=0; idx < points.size(); idx++)
```

gdmsl reviewed (changes requested):
Perfect. Seems good to me. Just fix the two things and then we merge.

galvanixer committed:
clang-format and minor modifications
1. Modifications suggested by Guido incorporated.
2. clang-format carried out for Bravais.hpp and BravaisTest.cpp
